### PR TITLE
Replace KoreBuild with Arcade

### LINKED ISSUE #88: Converge "KoreBuild" and Arcade
The way arcade builds, tests, packages, signs, and publishes must be directly usable by all repos in the stack.   Among other things, it provides the following value:
-  Consistency across repos for devs
-  Implements the "repo API", making adoption more straight forward
-  Allows for reasonable updates and improvements across the entire stack
-  More efficient, in that there's only one implementation for shared eng 

This epic is considered complete when:
-  [ ] Arcade has the changes necessary for KoreBuild repos
-  [ ] All KoreBuild driven changes are useful across the stack

### LINKED ISSUE #7740: AddAllProjectRefsToSolution fails for Middleware solution
### Describe the bug
The Middleware solution can't be opened due to the known issue NU1105, however running the AddAllProjectRefsToSolution.ps1 script doesn't work either.

### To Reproduce
1. Checkout master and clean your working copy
2. Run .\restore.cmd. This pass
3. Run ./eng/scripts/AddAllProjectRefsToSolution.ps1 -WorkingDir .\src\Middleware\
4. The script attempt to build but it fails. See error below

### Expected behavior
AddAllProjectRefsToSolution add all project references to the solution to circumvent the known issue, however it fails due to the same issue 

### Error
```powershell
❯ ./eng/scripts/AddAllProjectRefsToSolution.ps1 -WorkingDir .\src\Middleware\
Using KoreBuild 3.0.0-build-20190205.1
.NET Core SDK 3.0.100-preview-009750 is already installed. Skipping installation.
Detected Visual Studio Enterprise 2017 (15.9.28307.423) in 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise'
>>> msbuild.exe C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\build\tasks\RepoTasks.csproj -v:m -restore -t:Publish -p:Configuration=Release -p:PublishDir=C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\build\tasks\bin\publish -nologo -p:RepoTasksSdkPath=C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\.dotnet\buildtools\korebuild\3.0.0-build-20190205.1\scripts\..\msbuild\KoreBuild.RepoTasks.Sdk\Sdk /p:ProjectsReferencedOutFile=C:\Users\alessiof\AppData\Local\Temp\tmp9DBC.tmp /p:Projects=C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\**\*.*proj /p:_RunRestore=True /p:_RunBuild=True /p:_RunPack=False /p:_RunTests=False /p:_RunSign=False /p:TargetArchitecture=x64 /p:TargetOsName=win @C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\artifacts\logs\msbuild.logger.rsp
  Restore completed in 47.89 ms for C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\build\tasks\RepoTasks.csproj.
  RepoTasks -> C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\build\tasks\bin\Release\net461\RepoTasks.dll
  RepoTasks -> C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\build\tasks\bin\publish\

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.79
>>> msbuild.exe @C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\artifacts\logs\msbuild.rsp @C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\artifacts\logs\msbuild.logger.rsp
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices.Extensions\src\Microsoft.AspNetCore.SpaServices.Extensions.csproj : error NU1105: Unable to find p
roject information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, thi
s may be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices.Extensions\src\Microsoft.AspNetCore.SpaServices.Extensions.csproj : error NU1105: Unable to find p
roject information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this ma
y be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc\src\Microsoft.AspNetCore.Mvc.csproj'. Inside Visual Studio, this may be because the project is unloaded or not part of curre
nt solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, this may be because the project is unl
oaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this may be because the project is unloade
d or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\src\Microsoft.AspNetCore.SpaServices.csproj : error NU1105: Unable to find project information for
 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, this may be because the p
roject is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\src\Microsoft.AspNetCore.SpaServices.csproj : error NU1105: Unable to find project information for
 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this may be because the proje
ct is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\WebSockets\test\ConformanceTests\Microsoft.AspNetCore.WebSockets.ConformanceTests.csproj : error NU1105: Unabl
e to find project information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Servers\IIS\IntegrationTesting.IIS\src\Microsoft.AspNetCore.Server.IntegrationTesting.
IIS.csproj'. Inside Visual Studio, this may be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets requi
red for restore.

Build FAILED.

C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices.Extensions\src\Microsoft.AspNetCore.SpaServices.Extensions.csproj : error NU1105: Unable to find p
roject information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, thi
s may be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices.Extensions\src\Microsoft.AspNetCore.SpaServices.Extensions.csproj : error NU1105: Unable to find p
roject information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this ma
y be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc\src\Microsoft.AspNetCore.Mvc.csproj'. Inside Visual Studio, this may be because the project is unloaded or not part of curre
nt solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, this may be because the project is unl
oaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\samples\Webpack\Webpack.csproj : error NU1105: Unable to find project information for 'C:\Users\al
essiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this may be because the project is unloade
d or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\src\Microsoft.AspNetCore.SpaServices.csproj : error NU1105: Unable to find project information for
 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.ViewFeatures\src\Microsoft.AspNetCore.Mvc.ViewFeatures.csproj'. Inside Visual Studio, this may be because the p
roject is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\SpaServices\src\Microsoft.AspNetCore.SpaServices.csproj : error NU1105: Unable to find project information for
 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Mvc\Mvc.TagHelpers\src\Microsoft.AspNetCore.Mvc.TagHelpers.csproj'. Inside Visual Studio, this may be because the proje
ct is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Middleware\WebSockets\test\ConformanceTests\Microsoft.AspNetCore.WebSockets.ConformanceTests.csproj : error NU1105: Unabl
e to find project information for 'C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\src\Servers\IIS\IntegrationTesting.IIS\src\Microsoft.AspNetCore.Server.IntegrationTesting.
IIS.csproj'. Inside Visual Studio, this may be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets requi
red for restore.
    0 Warning(s)
    8 Error(s)

Time Elapsed 00:00:07.18
msbuild.exe failed with exit code: 1
At C:\Users\alessiof\Workspace\public\aspnet\AspNetCore\.dotnet\buildtools\korebuild\3.0.0-build-20190205.1\scripts\common.psm1:11 char:9
+         throw "$cmdName failed with exit code: $exitCode"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (msbuild.exe failed with exit code: 1:String) [], RuntimeException
    + FullyQualifiedErrorId : msbuild.exe failed with exit code: 1
```

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of #7280 

Resolves https://github.com/dotnet/arcade/issues/88

High level changes
* Update build.sh/ps1 to call into Arcade instead of KoreBuild. This completely removes our usage of KoreBuild
* Move all contents in build/ (the KoreBuild convention) into their matching version in eng/
* Remove dead code and other features which were KoreBuild only
  * Remove usage of NuGetPackageVerifier
  * Remove signcheck
  * Remove AddAllProjectRefsToSolution.ps1 (Resolves #7740 )

TODO (before this PR is ready to merge)
* [x] Get tests running 
* [x] Ensure we get the same build output
* [x] Ensure code gen steps still work
* [x] Consume a new update of Arcade which has necessary fixes
   * https://github.com/dotnet/arcade/issues/3006
   * https://github.com/dotnet/arcade/pull/3025

TODO (will be done in a separate PR)
~~* This removed support for generating Maestro manifests during build. We'll need to re-enable this somehow.~~ handled in this PR by keeping existing Maestro manifest generation
* Use the Arcade YAML templates for CI definition
* Fix site extensions build https://github.com/aspnet/AspNetCore/issues/11186
* Run sign check https://github.com/dotnet/arcade/issues/2888

### CODE REVIEW

File: build.ps1
Comment (dougbu) [inline build.ps1]: Add docs for this parameter

**Comment by natemcmaster** [comment]
Hmm, SiteExtensions are super broken by this. I'm looking at the targets which implement the site extension build and big chunks will probably need to be rewritten to work with Arcade.

@JunTaoLuo would you be okay temporarily disabling the site extension projects and doing that separately?

**Comment by JunTaoLuo** [comment]
Sure let's separate that work out of this PR. Let's discuss what the issues are.

**Comment by natemcmaster** [comment]
So, I've been spinning lots of builds trying to get this working. Here are the current things I'm stuck on:

* Tests hang indefinitely. This PR changes from using `dotnet vstest` to `xunit.console`. It appears xunit is hanging indefinitely
* Building Deb/RPM installers keeps failing with strange issues. We use `dockerbuild.sh` to build the deb/rpm projects in a container. With KoreBuild, this seemed to work fine. I'm not sure why changing to Arcade is causing so many issues -- can't find a clear correlation.

Also, these don't block merging, but need follow-up 

* Publishing with Arcade does not work due to https://github.com/dotnet/arcade/issues/3046. This PR continues to use our hacked approach to generating manifests and publishing (see eng/tools/Maestro/)
* Building site extensions fails. See list of reasons in https://github.com/aspnet/AspNetCore/issues/11186. Need to find workaround for Preview 7 because this PR disables building the site extension

**Comment by natemcmaster** [comment]
Oh yeah, one more:

* SignCheck has been removed. Aracde doesn't run signcheck during build. It's supposed to be done in the async publishing pipeline, but we can't use that yet either for two reasons https://github.com/dotnet/arcade/issues/2888 and https://github.com/dotnet/arcade/issues/3046. If we can't get async publishing working soon, we should consider re-adding signcheck as a manual step we run in our own pipeline.

File: eng/XUnitWithTimeout.targets
Comment (natemcmaster) [inline eng/XUnitWithTimeout.targets]: Hopefully this entire file can be removed before we merge. Added so I could add a timeout to xunit.

File: eng/XUnitWithTimeout.targets
Comment (natemcmaster) [inline eng/XUnitWithTimeout.targets]: Adding `Timeout="$(TestTimeoutMilliseconds)"` is the only change I made here when I copied this out of Arcade.

**Comment by natemcmaster** [comment]
I got this PR most of the way ready to merge. We're still seeing some test assemblies time out as a result of switching from `dotnet vstest` to `xunit.console`. @JunTaoLuo will be taking on finishing up the rest of this PR as I'm going to be on leave for several weeks.

**Comment by JunTaoLuo** [comment]
There are 4 test projects that are hanging.
```
/home/vsts/work/1/s/eng/XUnitWithTimeout.targets(93,5): warning MSB5002: Terminating the task executable "dash" because it did not finish within the specified limit of 900000 milliseconds. [/home/vsts/work/1/s/src/Hosting/Hosting/test/Microsoft.AspNetCore.Hosting.Tests.csproj]
/home/vsts/work/1/s/eng/XUnitWithTimeout.targets(93,5): warning MSB5002: Terminating the task executable "dash" because it did not finish within the specified limit of 900000 milliseconds. [/home/vsts/work/1/s/src/Hosting/TestHost/test/Microsoft.AspNetCore.TestHost.Tests.csproj]
/home/vsts/work/1/s/eng/XUnitWithTimeout.targets(93,5): warning MSB5002: Terminating the task executable "dash" because it did not finish within the specified limit of 900000 milliseconds. [/home/vsts/work/1/s/src/Security/Authentication/Negotiate/test/Negotiate.Test/Microsoft.AspNetCore.Authentication.Negotiate.Test.csproj]
/home/vsts/work/1/s/eng/XUnitWithTimeout.targets(93,5): warning MSB5002: Terminating the task executable "dash" because it did not finish within the specified limit of 900000 milliseconds. [/home/vsts/work/1/s/src/Mvc/test/Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]
```

I've been able to reproduce the hang with Microsoft.AspNetCore.Hosting.Tests.csproj. I can see in the artifact logs that the tests ran to completion. However, the exec command never completes.

**Comment by natemcmaster** [comment]
To unblock this, what if we temporarily skip the test assemblies that deadlock?

**Comment by JunTaoLuo** [comment]
I think I've fixed all the test timeouts now. They weren't hard, just really time consuming to find the culprits. However, now the test that weren't previously running are failing. Time to peel the (tests) onion.

File: src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs
Comment (JunTaoLuo) [inline src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs]: This caused a failure in xunit test reporting due to control characters appearing in logs, so I escaped the non-printable characters @halter73 Any objections to this change?

Comment (Tratcher) [reply src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs]: This looks good. Can you include a sample of the output?

Prior improvements here: https://github.com/aspnet/KestrelHttpServer/pull/3046/files

Comment (JunTaoLuo) [reply src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs]:
```
dbug: Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.LoggingConnectionAdapter[0]
      WriteAsync[129] 00 00 61 01 04 00 00 00 03 88 00 04 64 61 74 65 1D 4D 6F 6E 2C 20 31 37 20 4A 75 6E 20 32 30 31 39 20 32 32 3A 31 31 3A 32 31 20 47 4D 54 00 0C 63 6F 6E 74 65 6E 74 2D 74 79 70 65 0A 74 65 78 74 2F 70 6C 61 69 6E 00 06 73 65 72 76 65 72 07 4B 65 73 74 72 65 6C 00 0E 63 6F 6E 74 65 6E 74 2D 6C 65 6E 67 74 68 02 31 34 00 00 0E 00 00 00 00 00 03 68 65 6C 6C 6F 2C 20 77 6F 72 6C 64 0D 0A
      \x00\x00a\x01\x04\x00\x00\x00\x03\x88\x00\x04date\x1DMon, 17 Jun 2019 22:11:21 GMT\x00\x0Ccontent-type\x0Atext/plain\x00\x06server\x07Kestrel\x00\x0Econtent-length\x0214\x00\x00\x0E\x00\x00\x00\x00\x00\x03hello, world\x0D\x0A
```

Comment (Tratcher) [reply src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs]: That works. Was that an HTTP/2 response?

**Comment by JunTaoLuo** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=228611&view=logs.

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2623
https://github.com/aspnet/AspNetCore-Internal/issues/2410
https://github.com/aspnet/AspNetCore-Internal/issues/2415

**Comment by JunTaoLuo** [comment]
Tests are passing! Just need to verify build output before merging.

**Review by JunTaoLuo — APPROVED** [review]

**Review by jkotalik — APPROVED** [review]

File: Directory.Build.props
Comment (JunTaoLuo) [reply Directory.Build.props]: Maybe because it's what other repos do? https://github.com/aspnet/Extensions/blob/master/Directory.Build.props. Functionally doesn't make a difference though.

File: eng/AfterSolutionBuild.targets
Comment (JunTaoLuo) [inline eng/AfterSolutionBuild.targets]: Hmm I'm not sure if this is being used at all but it's probably important since @natemcmaster added it explicitly.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj]: Remove workaround.

**Comment by JunTaoLuo** [comment]
Just missing a signoff from @dougbu and another run of the tests to see if there are any flakiness.

**Comment by natemcmaster** [comment]
image

Nice work! Thanks for following through on this John!

File: Directory.Build.props
Comment (natemcmaster) [reply Directory.Build.props]: RepoRoot needs to be set before Arcade's Sdk.props file is imported.

File: eng/AfterSolutionBuild.targets
Comment (natemcmaster) [reply eng/AfterSolutionBuild.targets]: Yes, this used. This file is imported by Arcade and invoked as a part of build.cmd/sh

File: .azure/pipelines/helix-test.yml
Comment (dougbu) [inline .azure/pipelines/helix-test.yml]: Add `-noRestore` here

File: src/DefaultBuilder/test/Microsoft.AspNetCore.FunctionalTests/Microsoft.AspNetCore.FunctionalTests.csproj
Comment (dougbu) [inline src/DefaultBuilder/test/Microsoft.AspNetCore.FunctionalTests/Microsoft.AspNetCore.FunctionalTests.csproj]: Are these new references to projects not mentioned elsewhere showing up due to merges from 'master'?

File: src/Hosting/Hosting/test/WebHostBuilderTests.cs
Comment (dougbu) [inline src/Hosting/Hosting/test/WebHostBuilderTests.cs]: Lots 'o C# changes from here on also look odd. Are these also due to merges from 'master'?

**Review by dougbu — APPROVED** [review]
Generally looks good (though I'm not sure I really got everything in this huge PR)

File: src/Hosting/Hosting/test/WebHostBuilderTests.cs
Comment (JunTaoLuo) [reply src/Hosting/Hosting/test/WebHostBuilderTests.cs]: No this is to prevent a hang. Not disposing generic host builders caused this so I had to add using in several places. @Tratcher is this the issue tacking that bug https://github.com/aspnet/Extensions/issues/1363?

File: src/DefaultBuilder/test/Microsoft.AspNetCore.FunctionalTests/Microsoft.AspNetCore.FunctionalTests.csproj
Comment (JunTaoLuo) [reply src/DefaultBuilder/test/Microsoft.AspNetCore.FunctionalTests/Microsoft.AspNetCore.FunctionalTests.csproj]: I added these references to ensure build order between test assets and functional tests.

File: .azure/pipelines/helix-test.yml
Comment (JunTaoLuo) [reply .azure/pipelines/helix-test.yml]: Yea this should be safe.

File: .azure/pipelines/ci.yml
Comment (JunTaoLuo) [reply .azure/pipelines/ci.yml]: Hmm I might want to investigate this separately. Specifically because I know there were some changes with lifetimes of targets and interleaving of steps introduced with converting to arcade so these changes may be deliberate. Since I wouldn't want to hold up the PR and this is more of an optimization, I would do this as a follow up.

**Comment by JunTaoLuo** [comment]
If the next round of tests succeed, I'll conclude that these changes have not introduced any additional flakiness (a total of 3 successes) in our builds and will merge. I'll also file follow up issues.

**Comment by JunTaoLuo** [comment]
Followup items:
- https://github.com/aspnet/AspNetCore/issues/11359
- https://github.com/aspnet/AspNetCore/issues/11186
- https://github.com/aspnet/AspNetCore/issues/11360
- https://github.com/aspnet/AspNetCore/issues/11361

File: src/Hosting/Hosting/test/WebHostBuilderTests.cs
Comment (Tratcher) [reply src/Hosting/Hosting/test/WebHostBuilderTests.cs]: Correct

**Comment by jkotalik** [comment]
@JunTaoLuo I'll also make a follow-up PR to fix potential native dll copying issues. Let's get this in!

**Comment by dougbu** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=230108

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2489

**Comment by dougbu** [comment]
> Helix completed, not sure why the PR check isn't showing green/red.

We're still seeing the occasional missed messages between AzDO and GitHub. Since the Helix failure is of a known-flaky test, I suggest we get this in. @JunTaoLuo please do the honours.

**Pull request merged**
